core/src/main/java/io/aiven/klaw/config/MigrationUtility.java

```diff
@@ -200,4 +241,63 @@ private boolean isVersionGreaterThenCurrentVersion(
 
     return false;
   }
+
+  /**
+   * isNewInstall Checks the liquibase DatabaseChangeLog Table It checks if all the tables were
+   * created at the same time (by default all tables are created within an hour of each other)
+   *
+   * @return true if all tables are created at the same time, false if the database tables were
+   *     created over a longer period of time.
+   */
+  private boolean isNewInstall() {
+    try {
+      if (liquibase.getDataSource() != null) {
+        ResultSet results = getLiquibaseDatabaseChangeLogExecutionTimes();
+        LocalDateTime earliestTime = null, latestTime = null;
+
+        while (results.next()) {
+          LocalDateTime executionTime = results.getTimestamp(1).toLocalDateTime();
+
+          earliestTime = isExecutionTimeBeforeCurrentEarliestTime(earliestTime, executionTime);
+          latestTime = isExecutionTimeAfterCurrentLatestTime(latestTime, executionTime);
+        }
+        return earliestTime
```

Comment:
earliestTime can be null

Reply:
latestTime can be null

Reply:
Going to throw a runtime exception to stop Klaw running if this occurs. As we shouldnt proceed if we cant connect to the db.